composer/trainer/trainer.py

```diff
@@ -1645,7 +1645,7 @@ def _train_batch(self, use_grad_scaling: bool):
                         else:
                             optimizer.step()
             except RuntimeError as e:
-                if _is_cuda_oom(e):
+                if self.adaptive_gradient_accumulation and _is_cuda_oom(e):
```

Comment:
@ravi-mosaicml is it OK to do the check here? or should we always run the `should_handle_cuda_oom` logic, and then reraise it so that all the ranks fail?